lib/caching/NonceCache.ts

```diff
@@ -150,12 +150,20 @@ export class NonceCache {
   // We wait for a maximum of MAX_NONCE_LOCK_TIME_MS until a nonce lock is released
   private async waitForLockAndGetNonceInfo(
     signerAddress: string,
-    nonceType: string,
+    nonceType: NonceType,
     requestId: string
   ): Promise<CachedNonceInfo> {
     log.debug(new Date(), ` ${requestId} - Waiting for nonce to be unlocked. Max wait: ${MAX_NONCE_LOCK_TIME_MS}ms`);
-    for (let i = 0; i < Math.ceil(MAX_NONCE_LOCK_TIME_MS / NONCE_LOCK_POLL_INTERVAL_MS); i++) {
+    const maxIteration = Math.ceil(MAX_NONCE_LOCK_TIME_MS / NONCE_LOCK_POLL_INTERVAL_MS);
+
+    for (let i = 0; i < maxIteration; i++) {
       await Utils.sleep(NONCE_LOCK_POLL_INTERVAL_MS);
+
+      if (i + 1 === maxIteration) {
+        // If this is the last iteration, check if we can force unlock
+        await this.forceUnlockIfNonceIsExpired(signerAddress, nonceType, requestId);
+      }
```

Comment:
```suggestion
    for (let i = 0; i <= maxIteration; i++) {
      await Utils.sleep(NONCE_LOCK_POLL_INTERVAL_MS);

      if (i === maxIteration) {
        // If this is the last iteration, check if we can force unlock
        await this.forceUnlockIfNonceIsExpired(signerAddress, nonceType, requestId);
      }
```

Reply:
I don't think this suggestion is equivalent. The new suggestion will execute `maxIteration + 1` times instead of only `maxIteration`. 
The condition is just to free the lock on the last execution of the loop. Perhaps it will be more readable as
`if (i === maxIteration - 1)` ?

Or, then, move the forceUnlock to after the loop. If we exit the loop, then check if the lock is still on and expire it. May be clearer.